lib/requests/delete-account-confirm-request.js

```diff
@@ -0,0 +1,172 @@
+'use strict'
+
+const AuthRequest = require('./auth-request')
+const debug = require('./../debug').accounts
+const fs = require('fs-extra')
+
+class DeleteAccountConfirmRequest extends AuthRequest {
+  /**
+   * @constructor
+   * @param options {Object}
+   * @param options.accountManager {AccountManager}
+   * @param options.userStore {UserStore}
+   * @param options.response {ServerResponse} express response object
+   * @param [options.token] {string} One-time reset password token (from email)
+   */
+  constructor (options) {
+    super(options)
+
+    this.token = options.token
+    this.validToken = false
+  }
+
+  /**
+   * Factory method, returns an initialized instance of DeleteAccountConfirmRequest
+   * from an incoming http request.
+   *
+   * @param req {IncomingRequest}
+   * @param res {ServerResponse}
+   *
+   * @return {DeleteAccountConfirmRequest}
+   */
+  static fromParams (req, res) {
+    let locals = req.app.locals
+    let accountManager = locals.accountManager
+    let userStore = locals.oidc.users
+
+    let token = this.parseParameter(req, 'token')
+
+    let options = {
+      accountManager,
+      userStore,
+      token,
+      response: res
+    }
+
+    return new DeleteAccountConfirmRequest(options)
+  }
+
+  /**
+   * Handles a Change Password GET request on behalf of a middleware handler.
+   *
+   * @param req {IncomingRequest}
+   * @param res {ServerResponse}
+   *
+   * @return {Promise}
+   */
+  static get (req, res) {
+    const request = DeleteAccountConfirmRequest.fromParams(req, res)
+
+    return Promise.resolve()
+      .then(() => request.validateToken())
+      .then(() => request.renderForm())
+      .catch(error => request.error(error))
```

Comment:
The above is weird. Let's maybe just write as an `async` function.

Reply:
Ok, I can do this ^_^

Reply:
Great, @megoth ! I was hoping that you would step up to that, because I have just been doing what the rest of the code is doing. :-)